### ISSUE TITLE
Error when trying to get moderators

### ISSUE THREAD

**Issue by SushiWaUmai** [halcyonnouveau/roux#85]
I'm trying to collect all names of the moderators of a subreddit like this,
```rs
let mods: Vec<String> = sub
    .moderators()
    .await?.data.children
    .iter()
    .map(|x| String::from(&x.data.name))
    .collect();
```

but this returns me an error saying the following:

```
Error: error decoding response body: missing field `data` at line 1 column 38

Caused by:
    0: error decoding response body: missing field `data` at line 1 column 38
    1: missing field `data` at line 1 column 38
```

**Comment by beanpuppy** [comment]
Are you using the OAuth client ? The `moderators` endpoint requires authentication.

**Comment by beanpuppy** [comment]
Actually ignore that, the serde_json structure was wrong. This has been fixed with v2.2.12.
